docs/params.rst

```diff
@@ -0,0 +1,119 @@
+==========
+Parameters
+==========
+
+:ref:`Spider arguments <spiderargs>` have some limitations:
+
+-   There is no standard way for spiders to indicate which parameters they
+    expect or support.
+
+-   Since arguments can come from the command line, and those are always
+    strings, you have to write code to convert arguments to a different type
+    when needed.
+
+-   If you want argument validation, such as making sure that arguments are of
+    the right type, or that required arguments are present, you must implement
+    validation yourself.
+
+scrapy-spider-metadata allows overcoming those limitations.
+
+.. _define-params:
+
+Defining supported parameters
+=============================
+
+To define spider parameters, define a subclass of `pydantic.BaseModel`_,
+and then make your spider also inherit from
+:class:`~scrapy_spider_metadata.Args` with your parameter specification class
+as its parameter:
+
+.. _pydantic.BaseModel: https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel
+
+.. code-block:: python
+
+    from pydantic import BaseModel
+    from scrapy import Request, Spider
+    from scrapy_spider_metadata import Args
+
+    class MyParams(BaseModel):
+        pages: int
+
+    class MySpider(Args[MyParams], Spider):
+        name = "my_spider"
+
+        def start_requests(self):
+            for index in range(1, self.args.pages + 1):
```

Comment:
It'd be good to explain in the text how to access the parameters, and explain that they're going to be already validated & processed.

Reply:
Ah, I see, it's explaned below.